packages/glimmer-compiler/lib/compiler.ts

```diff
@@ -19,7 +25,15 @@ export type TemplateSpec = string;
  * @return {TemplateSpec} A template spec string
  */
 export function compileSpec(string: string, options: any): TemplateSpec {
+  let meta: BlockMeta = {};
+  if (typeof options === 'object') {
+    meta = Object.assign(meta, options.meta);
+  }
   let ast = preprocess(string, options);
   let program = TemplateCompiler.compile(options, ast);
-  return JSON.stringify(program);
+  let templateModule: TemplateModule = {
+    meta,
+    template: JSON.stringify(program)
+  };
+  return JSON.stringify(templateModule);
```

Comment:
@krisselden so you want the return type of this function changed from `String` to `Object`?